src/robot/conf/languages.py

```diff
@@ -67,36 +71,57 @@ def _get_languages(self, languages):
                 if normalized in available:
                     returned.append(available[normalized]())
                 else:
-                    returned.extend(self._import_languages(lang))
+                    returned.extend(v() for v in self.import_languages_module(lang).values())
         return returned
 
-    def _resolve_languages(self, languages):
+    def _resolve_languages(self, languages, add_default=True):
         if not languages:
             languages = []
         elif is_list_like(languages):
             languages = list(languages)
         else:
             languages = [languages]
-        languages.append(En())
+        if add_default:
+            languages.append(En())
+            # The English singular forms are added for backwards compatibility
+            self.headers = {
+                'Setting': 'Settings',
+                'Variable': 'Variables',
+                'Test Case': 'Test Cases',
+                'Task': 'Tasks',
+                'Keyword': 'Keywords',
+                'Comment': 'Comments'
+            }
         return languages
 
-    def _get_available_languages(self):
+    @staticmethod
+    def get_available_languages():
+        """Returns the currently available Languages as a dict of names and classes."""
         available = {}
         for lang in Language.__subclasses__():
             available[normalize(lang.__name__)] = lang
             if lang.__doc__:
                 available[normalize(lang.__doc__.splitlines()[0])] = lang
         return available
 
-    def _import_languages(self, lang):
+    @staticmethod
+    def import_languages_module(file_or_module):
+        """Imports a custom language module and returns the available languages."""
         def is_language(member):
             return (inspect.isclass(member)
                     and issubclass(member, Language)
                     and member is not Language)
-        if os.path.exists(lang):
-            lang = os.path.abspath(lang)
-        module = Importer('language file').import_module(lang)
-        return [value() for _, value in inspect.getmembers(module, is_language)]
+
+        if os.path.exists(file_or_module):
+            file_or_module = os.path.abspath(file_or_module)
+        file_or_module = Importer('language file').import_module(file_or_module)
+
+        available = {}
+        for _, lang in inspect.getmembers(file_or_module, is_language):
+            available[normalize(lang.__name__)] = lang
+            if lang.__doc__:
+                available[normalize(lang.__doc__.splitlines()[0])] = lang
+        return available
```

Comment:
What's the reason to return languages as a dict instead of a list? List would be easier for our code in `_get_languages` but is a dict better for external usage? Dict containing languages both by code (class name) and name (docstring) also feels a bit strange because in `_get_languages` we then add languages with both code and name twice.

If there's no strong reason to return a dict, I'd change this back to a list of instances. External users could then easily convert it to a dict if needed. Alternatively external users could first call this method and then `get_available_languages` that returns also languages imported here.

As a style issue, we haven't in general used empty lines inside methods or functions unless there are nested functions. If the code gets so long that empty lines are needed to split it to parts, we have added helper methods/functions instead.

Reply:
This issue is still open.